### ISSUE TITLE
Files backup of NX1

### ISSUE THREAD

**Issue by Tommixoft** [ottokiksmaler/nx500_nx1_modding#4]
Hi, sorry for opening issue i see no way other than thsi to communicate with you. Im Windows guy and suck at linux, 
In android we can make full backup of all partitions, is it possible to make backup of NX1 and extract contents so for example we could replace some of apps or settings on top of NX500. I think frame rates, bitrates are somewhere in configs or hardcoded into camera app.

Maybe you could clone NX1 (if you have it) and share somewhere? THANKS.

**Comment by ottokiksmaler** [comment]
Sorry for the late reply. 

All this is fairly well documented here and in linked threads on eoshd.com and dpreview.com.

TLDR version is that all images/partitions are in freely available firmware images and can be extracted by binwalk (as documented by Syme in eoshd.com). They are also there in /dev/mmc* files and can be copied by dd to sdcard (for example). 
The catch is that di-camera-app controls everything, it's closed source and we're trying to extend functionality by either giving it different commands by st, prefman and similar tools or debugging it via gdb (as documented here) and then trying to apply that in usable fashion. It's also linked to dozens of libraries. Everything is stripped of debugging symbols.
Another catch is that there are two sets of cpus and two independent operating systems running in camera, tizen and tronOS.  They communicate via mapped memory region. This is also completely undocumented. 
Everything I find out is documented here. 
I don't have nx1 so we're in the same position regarding that.

**Comment by ottokiksmaler** [comment]
Regarding bit rates, etc, Vasile should be back soon and he had demonstrated that he can alter the bit rate of video. 
You can as well, at breakpoint before the call to mmf-camcorder (IIRC, not near the computer) and alter the data being passed (they are strings in pairs, you're looking for video-encoder-bitrate and the following one). 
If they are hardcoded I cannot find them.

**Comment by Tommixoft** [comment]
wow you know a LOT! :D  Bitrates should be written somewhere for sure. ether in app or in some config. we need find mapping between lets say 2560_1440_30p and code who actually converts this defined variable to real resolution and bitrate and framerate. I will receive my nx500 tomorrow i guess and ill start digging too. sad im new to linux..

**Comment by ottokiksmaler** [comment]
To rip out rootfs, etc, read this post. I'm sure you can find equivalent tools for Windows (if not dd and generic decompressor) 
http://www.eoshd.com/comments/topic/19099-petition-for-samsung-nx1-hack/?do=findComment&comment=133037

**Comment by Tommixoft** [comment]
thanks i have ubuntu virtual machine installed so all good i know DD tool-yesterday tried to use on NX500 and i got total shit :D but know when i have offsets and sizes -it's good to go. THANKS A LOT!

**Comment by ottokiksmaler** [comment]
Use binwalk and look for LZMA (all filesystem are compressed in firmware file)
